src/Stratis.Bitcoin/Configuration/NodeSettings.cs

```diff
@@ -116,34 +146,24 @@ public NodeSettings(Network innerNetwork = null, ProtocolVersion protocolVersion
         /// <param name="network">Specification of the network the node runs on - regtest/testnet/mainnet.</param>
         /// <param name="protocolVersion">Supported protocol version for which to create the configuration.</param>
         /// <returns>Default node configuration.</returns>
-        public static NodeSettings Default(Network network = null, ProtocolVersion protocolVersion = SupportedProtocolVersion)
+        public static NodeSettings Default(Network network = null, ProtocolVersion protocolVersion = SupportedProtocolVersion, string[] args = null)
         {
-            NodeSettings nodeSettings = new NodeSettings(innerNetwork: network);
-            nodeSettings.LoadArguments(new string[0]);
-            return nodeSettings;
+            return new NodeSettings(network, protocolVersion, args:args);
         }
 
         /// <summary>
-        /// Initializes configuration from command line arguments.
-        /// <para>This includes loading configuration from file.</para>
+        /// Loads the configuration file.
         /// </summary>
-        /// <param name="args">Application command line arguments.</param>
         /// <returns>Initialized node configuration.</returns>
         /// <exception cref="ConfigurationException">Thrown in case of any problems with the configuration file or command line arguments.</exception>
-        public NodeSettings LoadArguments(string[] args)
+        public NodeSettings LoadConfiguration()
         {
-            // By default, we look for a file named '<network>.conf' in the network's data directory,
-            // but both the data directory and the configuration file path may be changed using the -datadir and -conf command-line arguments.
-            this.ConfigurationFile = args.GetValueOf("-conf")?.NormalizeDirectorySeparator();
-            var dataDir = args.GetValueOf("-datadir")?.NormalizeDirectorySeparator();
+            // Configuration already loaded?
+            if (this.ConfigReader != null)
+                return this;
 
-            // If the configuration file is relative then assume it is relative to the data folder and combine the paths
-            if (dataDir != null && this.ConfigurationFile != null)
-            {
-                bool isRelativePath = Path.GetFullPath(this.ConfigurationFile).Length > this.ConfigurationFile.Length;
-                if (isRelativePath)
-                    this.ConfigurationFile = Path.Combine(dataDir, this.ConfigurationFile);
-            }
+            // Get the arguments set previously
+            var args = this.LoadArgs ?? new string[] { };
```

Comment:
Will LoadArgs ever be null here?

Reply:
Not possible anymore. Removing the check.